packages/remix-react/components.tsx

```diff
@@ -1161,7 +1161,9 @@ export function useSubmitImpl(key?: string): SubmitFunction {
 
         // Include name + value from a <button>
         if (target.name) {
-          formData.append(target.name, target.value);
+          if (!formData.getAll(target.name).includes(target.value)) {
```

Comment:
```suggestion
          if (!formData.has(target.name)) {
```

That should work right?

Reply:
Nop, because it would reset https://github.com/remix-run/remix/commit/5894598c75d0e542651dc92bc400320c108eae8d effort.

```
<Form ...>
  <input type="hidden" name="somethingIWantToKeep" value="1" />
  <button type="submit" name="somethingIWantToKeep" value="2" />
</Form>
```

I don't know if there's a lot of use case for that, but @nrako has fixed it for a reason.

Reply:
Like `!formData.has` + `formData.append` is almost equal to `formData.set`, in the sens that, if your form already have an entry with your form button's name then, (in Chrome + Firefox and maybe others) your button's name + value will not be added. In Safari it's different because the button with it's name + value is already added in the form's data.

So the easiest work around is to check if there's already a similar entry in your form data. Since we can't check from what input comes the entry in the form data class.

It's not perfect, and maybe we don't want to support this use case (having a button with the exact same name of inputs) but in this case we should revert the pr from @nrako.

Reply:
Right gotcha, honestly the use case feels weird for duping name attribute, would be good to better understand that.

Reply:
There's a legitimate use case of having the button name attribute in the form data. Such as having multiple actions possibles in the remix's action.

```
<Form ...>
  <input type="hidden" name="id" value="1" />
  <button type="submit" name="intent" value="delete">Delete Todo</button>
</Form>

<Form ...>
  <input type="text" name="todo" />
  <button type="submit" name="intent" value="add">Add Todo</button>
</Form>
```

For the @nrako use case, I think we should refer to its commit but I can understand that erasing former inputs at the `set` stage could be a little bit annoying, even if I arguably don't see any benefits over hidden input or just changing the name attribute.

So we have a clear choice, either : 
- Revert  https://github.com/remix-run/remix/commit/5894598c75d0e542651dc92bc400320c108eae8d commit in order to avoid duplicated form data entry
- Merge this fix in order to fix duplicated entry on Safari

Reply:
In that case you would only get either set of formData right? Never both at the same time?

Reply:
I’ m not sure to understand. 
On the example I gave, it was just to demonstrate the need of having the button name / value in form data.

in this example, on safari, the intent has two intent / delete or intent / add (depending on where you click.

For the commit of @nrako I let you refer to its commit to understand his need.